### PR TITLE
Story 4 - Validation logic

### PR DESCRIPTION
**Pull request by cdc895642**
add validation when create new Group

### CODE REVIEW

File: src/main/java/com/softserve/teamproject/entity/Group.java [hunk 1]
```diff
@@ -162,17 +175,17 @@ public int hashCode() {
...
+  //        + ", name='" + name + '\''
+  //        + ", teachers=" + teachers
+  //        + ", location=" + location
+  //        + ", startDate=" + startDate
+  //        + ", finishDate=" + finishDate
+  //        + ", status=" + status
+  //        + ", specialization=" + specialization
+  //        + '}';
+  //  }
 }
```
Comment (thedikiy) [inline src/main/java/com/softserve/teamproject/entity/Group.java]: why did you comment out this?

File: src/main/java/com/softserve/teamproject/entity/Group.java [hunk 2]
```diff
@@ -47,13 +58,15 @@
...
+  @NotNull
   private Specialization specialization;
 
   @ElementCollection
   @CollectionTable(name = "expert",
           joinColumns = @JoinColumn(name = "edu_group_id")
   )
   @Column(name = "expert_name")
+  @StringConstraintInSet(min=5, max=25, regexp = "[\\p{IsAlphabetic}\\p{IsWhite_Space}-\\.]+")
   private Set<String> experts;
```
Comment (thedikiy) [inline src/main/java/com/softserve/teamproject/entity/Group.java]: I think 25 can be not enough

File: src/main/java/com/softserve/teamproject/entity/Group.java (hunk 1 above)
Comment (cdc895642) [inline src/main/java/com/softserve/teamproject/entity/Group.java]: StackOwerflowError

Comment (svoronin) [inline src/main/java/com/softserve/teamproject/entity/Group.java]: It doesn't seem to be the best approach of solving issues like that. I wouldn't recommend merging that in master

Comment (cdc895642) [inline src/main/java/com/softserve/teamproject/entity/Group.java]: fixed

Comment (svoronin) [inline src/main/java/com/softserve/teamproject/entity/Group.java]: tks

**Pull request merged**